src/fileparse.py

```diff
@@ -0,0 +1,88 @@
+import csv
+import logging
+from pathlib import Path
+from typing import Any, List, TextIO, Union
+
+
+def parse_csv(
+    file_path_or_buffer: Union[str, Path, TextIO],
+    select: List[str] = None,
+    convert_fn: Any = None,  # Using Union[List[Callable], Dict[str, Callable]] breaks everything :(
+    has_headers: bool = True,
+    delimiter: str = ",",
+    verbose: bool = True,
+) -> List[Any]:
+    """
+    Parse a csv file into a list.
+
+    :param file_path_or_buffer: Path to the csv file or file-like object
+    :param select: Columns to select
+    :param convert_fn: List or dictionary of functions to apply to each column
+    :param has_headers: Indicator if file has headers.
+                        If False, creates a list of tuples otherwise a list of dictionaries
+    :param delimiter: Type of delimiter. Defaults to ','
+    :param verbose: Indicates if errors should be logged
+    :raises RuntimeError: Raises RuntimeError if arguments are not compatible with each other.
+                          If select is given then has_headers must be True.
+                          Keys in convert_fn must be the same values in select.
+                          If there are headers then convert_fn must be a dict of col_name:type.
+                          Keys in convert_fn must match the name of the headers.
+    :return: List of items in the file
+    """
+
+    if not has_headers and select:
+        raise RuntimeError("'select' argument requires column headers")
+    if (select and convert_fn) and (select != list(convert_fn.keys())):
```

Comment:
The parenthesis is not necessary here and the final condition is only applicable when `convert_fn` is a dict, in which case it would be better to do a set comparison between the columns in `select` and the column keys in `convert_fn`

```suggestion
    if select and convert_fn and isinstance(convert_fn, dict) and set(select) != set(convert_fn.keys()):
```

Reply:
Changed.